identity-service/src/routes/healthCheck.js

```diff
@@ -76,29 +95,70 @@ module.exports = function (app) {
       // If the max number of transactions have been evaluated, break out
       if (txCounter > maxTransactions) break
       let block = await web3.eth.getBlock(i, true)
+      if (!minBlockTime || block.timestamp < minBlockTime) minBlockTime = block.timestamp
       if (block && block.transactions.length) {
         for (const tx of block.transactions) {
           // If transaction is from audius account, determine success or fail status
           if (RELAY_HEALTH_ACCOUNT === tx.from) {
             const txHash = tx.hash
             const resp = await web3.eth.getTransactionReceipt(txHash)
             txCounter++
-            if (!resp.status) failureTxHashes.push(txHash)
+
+            // tx failed
+            if (!resp.status) {
+              const senderAddress = await redis.hget('txHasToSenderAddress', txHash)
+              if (senderAddress) {
+                if (!failureTxs[senderAddress]) failureTxs[senderAddress] = [txHash]
+                else failureTxs[senderAddress].push(txHash)
+              } else failureTxs['unknown'].push(txHash)
+            }
           }
         }
       }
     }
 
+    let isError = false
+    if (Object.keys(failureTxs).length >= RELAY_HEALTH_MIN_NUM_USERS &&
+      flatten(Object.values(failureTxs)).length > maxErrors) isError = true
+
+    // delete old entries from set in redis
+    const epochOneHourAgo = Math.floor(Date.now() / 1000) - 3600
```

Comment:
looks like the MAX_BLOCK_RANGE = 200, which is ~17min, so is this cache eviction just set to be some amount larger than that? but also i think you have set a ceiling on how old the startBlock can be, so this just needs to be earlier than the earliest possible startBlock?

Reply:
so cache eviction is set to all items before 1 hour ago. it is technically possible to set a start and end block override greater than an hour ago. so maybe what we do here is force endBlock to always be the latest block. Max time you can look back is 17 minutes, but it caches for 1 hour

Reply:
endBlock = latest block fixes a ton of this complexity, sounds good